alica_engine/src/engine/BasicBehaviour.cpp

```diff
@@ -66,8 +66,26 @@ void BasicBehaviour::doRun()
     }
 }
 
+std::string BasicBehaviour::resultToString(BehResult result)
+{
+    switch (result) {
+    case BehResult::SUCCESS:
+        return "Success";
+    case BehResult::FAILURE:
+        return "Failure";
+    case BehResult::UNKNOWN:
+        return "Unknown";
+    }
+    return ""; // should never reach
+}
+
 void BasicBehaviour::doTerminate()
 {
+    if (getTrace()) {
```

Comment:
Is there a possibility we don't have a trace at this point?

Reply:
I believe if trace was not created in the first place, this check was also in setResult. I don't yet posses full initialization process knowledge

Reply:
Ah makes sense.  For each plan/behavior we can configure tracing to be skipped